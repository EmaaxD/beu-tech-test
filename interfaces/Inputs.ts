export interface SimpleInputProps {
  title: string;
  name: string;
  value: string;
  placeholder?: string;
  onHandleChange: (target: any) => void;
}

export interface SimpleTextareaProps {
  title: string;
  name: string;
  value: string;
  placeholder?: string;
  onHandleChange: (target: any) => void;
}

export interface StateReviewForm {
  username: string;
  review: string;
}
