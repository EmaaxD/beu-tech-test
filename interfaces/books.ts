export interface BookCardProps {
  id: string | number;
  image: string;
  title: string;
  authors: string;
  noDescription?: boolean;
  description: string;
  bookSM?: boolean;
  fullInfo?: boolean;
}
