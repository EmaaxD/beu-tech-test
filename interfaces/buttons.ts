export interface MainButtonProps {
  type?: ButtonTypes;
  text: string;
  disabled: boolean;
}

export interface MainIconProps {
  image: string;
  onHandleClick?: (value: any) => void;
}

export interface ModalButtonProps {
  type?: ButtonTypes;
  actions: ButtonActions;
  text: string;
  onHandleClick: (value: any) => void;
}

export type ButtonTypes = "button" | "submit" | "reset";

export type ButtonActions = "delete" | "cancel";
