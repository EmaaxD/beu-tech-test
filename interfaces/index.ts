export * from "./Inputs";
export * from "./reviews";
export * from "./books";
export * from "./buttons";
export * from "./services";
