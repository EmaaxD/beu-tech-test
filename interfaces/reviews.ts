export interface Reviews {
  id: string;
  user: string;
  comment: string;
  createdAt: string;
}

export interface ReviewCardProps {
  id: string;
  user: string;
  comment: string;
  createdAt: string | Date;
  onHandleReviewEdit: (value: any) => void;
  onHandleDltReview: (value: any) => void;
}

export interface ReviewsLS {
  id: string | any;
  comments: Reviews[];
}
