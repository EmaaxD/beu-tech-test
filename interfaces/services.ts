export interface ResponseGoogleApi {
  kind: string;
  totalItems: number;
  items: Item[];
}

export interface BookDataProps {
  id: string;
  title: string;
  authors: string;
  image: string;
  description: string;
}

export interface Item {
  kind: Kind;
  id: string;
  etag: string;
  selfLink: string;
  volumeInfo: VolumeInfo;
  saleInfo: SaleInfo;
  accessInfo: AccessInfo;
  searchInfo: SearchInfo;
}

export interface AccessInfo {
  country: Country;
  viewability: Viewability;
  embeddable: boolean;
  publicDomain: boolean;
  textToSpeechPermission: TextToSpeechPermission;
  epub: Epub;
  pdf: Epub;
  webReaderLink: string;
  accessViewStatus: AccessViewStatus;
  quoteSharingAllowed: boolean;
}

export enum AccessViewStatus {
  None = "NONE",
  Sample = "SAMPLE",
}

export enum Country {
  Ar = "AR",
}

export interface Epub {
  isAvailable: boolean;
  acsTokenLink?: string;
}

export enum TextToSpeechPermission {
  Allowed = "ALLOWED",
}

export enum Viewability {
  NoPages = "NO_PAGES",
  Partial = "PARTIAL",
}

export enum Kind {
  BooksVolume = "books#volume",
}

export interface SaleInfo {
  country: Country;
  saleability: Saleability;
  isEbook: boolean;
  listPrice?: SaleInfoListPrice;
  retailPrice?: SaleInfoListPrice;
  buyLink?: string;
  offers?: Offer[];
}

export interface SaleInfoListPrice {
  amount: number;
  currencyCode: CurrencyCode;
}

export enum CurrencyCode {
  Ars = "ARS",
}

export interface Offer {
  finskyOfferType: number;
  listPrice: OfferListPrice;
  retailPrice: OfferListPrice;
}

export interface OfferListPrice {
  amountInMicros: number;
  currencyCode: CurrencyCode;
}

export enum Saleability {
  ForSale = "FOR_SALE",
  NotForSale = "NOT_FOR_SALE",
}

export interface SearchInfo {
  textSnippet: string;
}

export interface VolumeInfo {
  title: string;
  subtitle?: string;
  authors: string[];
  publisher: string;
  publishedDate: Date;
  description: string;
  industryIdentifiers?: IndustryIdentifier[];
  readingModes: ReadingModes;
  pageCount: number;
  printType: PrintType;
  maturityRating: MaturityRating;
  allowAnonLogging: boolean;
  contentVersion: ContentVersion;
  panelizationSummary?: PanelizationSummary;
  imageLinks: ImageLinks;
  language: Language;
  previewLink: string;
  infoLink: string;
  canonicalVolumeLink: string;
  categories?: string[];
  averageRating?: number;
  ratingsCount?: number;
}

export enum ContentVersion {
  Preview100 = "preview-1.0.0",
  The1110Preview3 = "1.1.1.0.preview.3",
  The1120Preview3 = "1.1.2.0.preview.3",
}

export interface ImageLinks {
  smallThumbnail: string;
  thumbnail: string;
}

export interface IndustryIdentifier {
  type: Type;
  identifier: string;
}

export enum Type {
  Isbn10 = "ISBN_10",
  Isbn13 = "ISBN_13",
}

export enum Language {
  En = "en",
  Es = "es",
}

export enum MaturityRating {
  NotMature = "NOT_MATURE",
}

export interface PanelizationSummary {
  containsEpubBubbles: boolean;
  containsImageBubbles: boolean;
}

export enum PrintType {
  Book = "BOOK",
}

export interface ReadingModes {
  text: boolean;
  image: boolean;
}
