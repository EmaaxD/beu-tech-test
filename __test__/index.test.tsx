import React from "react";
import { RecoilRoot } from "recoil";
import { render, screen } from "@testing-library/react";

import { BookDataProps } from "@/interfaces/index";
import Home from "@/pages/index";

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/",
      pathname: "/",
      query: {},
      asPath: "/",
      basePath: "/",
      isLocaleDomain: true,
      isReady: true,
      push: jest.fn(),
      prefetch: jest.fn(),
      replace: jest.fn(),
      reload: jest.fn(),
      back: jest.fn(),
      beforePopState: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
        emit: jest.fn(),
      },
      isFallback: false,
      isPreview: false,
    };
  },
}));

describe("Home", () => {
  const books: BookDataProps[] = [
    {
      id: "asdasd",
      authors: "Anonymous",
      image:
        "https://jackmoreno.files.wordpress.com/2014/04/cantar-del-mc3ado-cid.jpg?w=648",
      description: "this is a simple descripcion for a book",
      title: "react testing",
    },
  ];

  test("<Home /> empieza a montarse", () => {
    render(
      <RecoilRoot>
        <Home books={books} />
      </RecoilRoot>
    );

    expect(screen.getByTestId("logo-app").tagName).toBe("IMG");

    expect(screen.getByTestId("input-search").tagName).toBe("DIV");

    expect(screen.getByTestId("title-footer").tagName).toBe("H6");

    expect(screen.getByTestId("container-main-imgs").children.length).toEqual(
      1
    );
  });
});
