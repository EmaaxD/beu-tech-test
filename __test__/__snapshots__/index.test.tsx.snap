// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<DocumentFragment>
  <div
    class="css-mgq3m7-MuiStack-root"
  >
    <div
      style="opacity: 0; transform: scale(0.5) translateZ(0);"
    >
      <header
        class="css-nen11g-MuiStack-root"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27180%27%20height=%2770%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="main logo app"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
      </header>
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
    >
      <div
        style="opacity: 0; transform: scale(0.5) translateZ(0);"
      >
        <form
          class="css-8flg95"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                  data-testid="SearchOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </div>
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
                id="search"
                name="search"
                placeholder="Buscar libro"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1 css-1sl0s6e-MuiTypography-root"
            />
          </div>
        </form>
      </div>
    </div>
    <main>
      <div
        class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
      >
        <div
          class="css-1u0a2d3-MuiStack-root"
        >
          <div
            style="opacity: 0; transform: scale(0.5) translateZ(0);"
          >
            <div
              class="css-2nb6pb-MuiStack-root"
            >
              <div
                class="MuiBox-root css-1aple4v"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27450%27%20height=%27700%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="book cover"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <span
                aria-label="4 Stars"
                class="MuiRating-root MuiRating-sizeMedium MuiRating-readyOnly css-ryrseu-MuiRating-root"
                color="#DEC700 !important"
                role="img"
              >
                <span>
                  <span
                    class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                </span>
                <span>
                  <span
                    class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                </span>
                <span>
                  <span
                    class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                </span>
                <span>
                  <span
                    class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                </span>
                <span>
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
              <div
                class="css-q95hb8-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ai5u1v-MuiTypography-root"
                >
                  react test
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahje9m-MuiTypography-root"
                >
                  Anonymous
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-7v0d5e"
        />
      </div>
    </main>
    <div
      class="MuiBox-root css-xi606m"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
      >
        BEU technical test - Emanuel Mamani 🚀
      </h6>
    </div>
  </div>
</DocumentFragment>
`;
